oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/JournalReader.java

```diff
@@ -54,7 +55,13 @@ protected JournalEntry computeNext() {
             String line = null;
             while ((line = reader.readLine()) != null) {
                 if (line.indexOf(' ') != -1) {
-                    List<String> splits = Splitter.on(' ').splitToList(line);
+                    List<String> splits;
+                    if (line.trim().isEmpty()) {
+                        // special case handling for empty lines where we need to split by space
+                        splits = Arrays.stream(line.split("")).map(String::trim).collect(Collectors.toList());
+                    } else {
+                        splits = Arrays.stream(line.split(" ")).collect(Collectors.toList());
+                    }
```

Comment:
I do not think that the code does still the same in case the trimmed line is empty.

Reply:
thanks for @mbaedke, could you please the test cases that you tried.
Also, this class has extensive test coverage and they all passed after these changes, but still we might have missed few cases, would be more than happy to add them.

Reply:
This class must parse a file that contains either three empty spaces on a line or a line with three entries separated by a space. Moreover, the unit cases provide extensive coverage for that.

Reply:
The output was:

```
import org.apache.jackrabbit.guava.common.base.Splitter
import java.util.Arrays
import java.util.List
import java.util.stream.Collectors

field String separator = " "

Defined field String input = "   "

Defined field List<String> guavaList = [, , , ]

System.out.println("Guava: " + guavaList.size());
Guava: 4


Defined field List<String> jseList = [, , ]

System.out.println("Stream API: " + jseList.size());
Stream API: 3

```

Reply:
Since the lists are of different length, I didn't look further.